client/update.go

```diff
@@ -185,6 +189,10 @@ func (c *Channel) update(ctx context.Context, next *channel.State) (err error) {
 
 	pidx, res, err := resRecv.Next(ctx)
 	if err != nil {
+		// Context expiry here implies we have timedout waiting for response from the peer.
+		if pcontext.IsContextError(err) {
+			return errors.WithStack(PeerNotRespondingError("peer did not respond to the channel proposal"))
```

Comment:
This function still uses `err` in a defer in line 161, better set it to the newly created error here.